raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -1391,10 +1436,61 @@ def handle_lock_expired(
     return TransitionResult(mediator_state, events)
 
 
+def handle_node_change_network_state(
+        mediator_state: MediatorTransferState,
+        state_change: ActionChangeNodeNetworkState,
+        channelidentifiers_to_channels: ChannelMap,
+        pseudo_random_generator: random.Random,
+        block_number: BlockNumber,
+) -> TransitionResult:
+    """ If a certain node comes online:
+    1. Check if a channel exists with that node
+    2. Check that this channel is a route, check if the route is valid.
+    3. Check that the transfer was stuck because there was no route available.
```

Comment:
Question: This bullet point refers to `if mediator_state.waiting_transfer is None:` ?

Reply:
Yes, in case the mediator has a `waiting_transfer`, then it makes sense for it to act upon receiving a notification about the next hop becoming available to try to transmit the transfer again.